PublicGitArchive/pga-create/process.go

```diff
@@ -454,6 +466,58 @@ func sivaFiles(inits map[model.SHA1]struct{}) []string {
 	return files
 }
 
+func sivaSize(init string) (int64, error) {
+	info, err := ioutil.ReadDir("/tmp/sourced")
+	if err != nil {
+		return -1, err
+	}
+
+	if len(info) != 1 || !info[0].IsDir() {
+		return -1, fmt.Errorf("/tmp/sourced directory wasn't in a clean status")
+	}
+
+	tmp := filepath.Join("/tmp/sourced", info[0].Name(), "transactioner")
+	info, err = ioutil.ReadDir(tmp)
+	if err != nil {
+		return -1, err
+	}
+
+	// regex to match directory names composed by sha1_timestamp as in
+	// 7a80dfe1684664cefd2923bdbb329dcb9a48dc4f_1551878586555302343
+	reg := regexp.MustCompile(`\b([0-9a-f]{40})_[0-9]{19}\b`)
+	var path string
+	for _, fi := range info {
+		if !fi.IsDir() {
+			continue
+		}
+
+		matches := reg.FindStringSubmatch(fi.Name())
+		if len(matches) != 2 {
+			continue
+		}
+
+		if init == matches[1] {
```

Comment:
As there can theoretically be more than one siva file and they could be still copying don't stop at the first match. Do the stat here and use the biggest size.

```go
if fi.Size() > size {
  size = fi.Size()
}
```